virtual-desktop/src/app/application-manager/application-manager.service.ts

```diff
@@ -51,17 +51,45 @@ export class ApplicationManager implements MVDHosting.ApplicationManagerInterfac
         return this.spawnApplication(plugin as DesktopPluginDefinitionImpl, metadata);
       });
     });
+    window.addEventListener('message',(message:any)=> {
+      if (message.data === 'iframeload') {
+        this.applicationInstances.forEach((appInstance)=> {
+          let instance = appInstance.instanceId;
+          const iframe:HTMLElement|null = document.getElementById(`${IFRAME_NAME_PREFIX}${instance}`);
+          console.log(`iframe=`,iframe);
+          if (iframe) {
+            if ((iframe as any).contentWindow == message.source
+                || (iframe as any).contentWindow == message.source.parent
+                || ((iframe as any).src.indexOf(message.origin) == 0)) {
+              //it's this one
+              const appInstance = this.applicationInstances.get(instance);
+              if (appInstance) {
+                const pluginId = appInstance.plugin.getIdentifier();
+                this.logger.info(`Iframe loaded: ${pluginId}, instance=${instance}`);
+                return ZoweZLUX.dispatcher.iframeLoaded(instance, pluginId);
+              }
+            }
+          }
+        });
+        this.logger.warn(`No iframe identified as source of message`);
+      }
+    });
     (window as any).ZoweZLUX.dispatcher.setPostMessageHandler( (instanceId:MVDHosting.InstanceId, message:any ) => {
        let applicationInstance:ApplicationInstance|undefined = this.applicationInstances.get(instanceId);
        if (applicationInstance){
-         this.logger.debug(`PostMessage for instance=${applicationInstance}, iframeID=${applicationInstance.iframeId}`);
-         let theIframe:HTMLElement|null = document.getElementById(applicationInstance.iframeId);
+         let theIframe:HTMLElement|null = document.getElementById(`${IFRAME_NAME_PREFIX}${instanceId}`);
          if (theIframe){
-           this.logger.debug(`PostMessage iframe found=`,theIframe);
+           this.logger.debug(`Checking if double-iframe`);
+           let secondIframe = (theIframe as HTMLIFrameElement).contentWindow.document.getElementById('zluxIframe');
```

Comment:
Leave a code comment to explain or make a constant about the fact that this is a reserved secret iframe keyword